schemas/json/layout/layout.schema.v1.json

```diff
@@ -0,0 +1,213 @@
+            "title": "dataModelBindings",
+            "description": "Data model bindings for a component.",
+            "additionalProperties": {
+              "type": "string"
+            }
+          }
+        },
+        "required": ["id", "type"],
+        "allOf": [
+          { "if": {"properties": {"type": { "const": "Datepicker"}}}, "then": { "$ref": "#/definitions/datepickerComponent"}},
+          { "if": {"properties": {"type": { "const": "FileUpload"}}}, "then": { "$ref": "#/definitions/fileUploadComponent"}},
+          { "if": {"properties": {"type": { "const": "NavigationButtons"}}}, "then": { "$ref": "#/definitions/navigationButtonsComponent"}},
+          { "if": {"properties": {"type": { "const": "Group"}}}, "then": { "$ref": "#/definitions/groupComponent"}},
+          { "if": {"properties": {"type": { "const": "Dropdown"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "RadioButtons"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "Checkboxes"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "AddressComponent"}}}, "then": { "$ref": "#/definitions/addressComponent"}}
+        ]
+      },
+      "fileUploadComponent": {
+        "properties": {
+          "maxFileSizeInMB": {
+            "title": "maxFileSizeInMB",
+            "description": "Sets the maximum file size allowed in megabytes.",
+            "type": "integer"
+          },
+          "maxNumberOfAttachments": {
+            "title": "maxNumberOfAttachments",
+            "description": "Sets the maximum number of attachments allowed to upload.",
+            "type": "integer"
+          },
+          "minNumberOfAttachments": {
+            "title": "minNumberOfAttachments",
+            "description": "Sets the minimum number of attachments to upload",
+            "type": "integer"
+          },
+          "displayMode": {
+            "title": "displayMode",
+            "description": "Sets the display mode for the file upload component.",
+            "type":"string",
+            "enum": ["simple", "list"]
+          },
+          "hasCustomFileEndings": {
+            "title": "hasCustomFileEndings",
+            "description": "Boolean value indicating if the component has valid file endings",
+            "type": "boolean"
+          },
+          "validFileEndings": {
+            "title": "validFileEndings",
+            "description": "A seperated string of valid file endings to upload. If not set all endings are accepted.",
+            "type": "string"
+          }
+        },
+        "required": ["maxFileSizeInMB", "maxNumberOfAttachments", "minNumberOfAttachments", "displayMode" ]
+      },
+      "datepickerComponent": {
+        "properties": {
+          "minDate": {
+            "type": "string",
+            "title": "minDate",
```

Comment:
I guess titles should not always be the same as the prop name. This could be "Minimum Allowed Date" for example.
See example: https://github.com/Altinn/altinn-cdn/blob/master/schemas/json/policy/policy.schema.v1.json#L33

This makes intellisense/help in vscode better and more readable.